docs/guides/misc/setting-static-ip.md

```diff
@@ -0,0 +1,78 @@
+[DRAFT]
+
+When you plug your device into the network, you will be assigned an IP address from the router. For most use cases, this is all that is needed, but for servers it is strongly recommended that the IP address is made static.
+
+This means ensuring that no matter how many times the device is rebooted / disconnected and reconnected it will _always_ get the same IP address.
+
+Depending on your operating system / device. There are many ways to achieve this, we will list a few here.
+
+# Raspbian / Raspberry Pi OS
+
+First, lets make take some notes about our current network settings with the following command:
+
+```bash
+ip r | grep default
+```
+
+This will spit out something along the lines of:
+
+```text
+default via 192.168.1.254 dev eth0 proto dhcp src 192.168.1.80 metric 202`
+```
+
+The first IP address here (`192.168.1.254`) is the IP address of our router. The second one (`192.168.1.80`) is the IP address that the router has assigned our Raspberry pi. Depending on your requirements, you can either choose to keep this IP or change it entirely. Just make sure that the IP address you choose is not already in use.
+
+We also need to know the DNS server that has been set. to do this run the following command:
+
+```bash
+cat /etc/resolv.conf
+```
+
+Which will output something along the lines of:
+
+```
+# Generated by resolvconf
+nameserver 192.168.1.254
+```
+
+We are interested in the IP address next to `nameserver`.
+
+Now that you have the above IP addresses noted down, it's time to modify `dhcpcd.conf` to set it as static. To do so, first run this command:
```

Comment:
I though about adding Debian's default `ifupdown`/`/etc/network/interfaces` based setup. However, until here everything is the same, also e.g. on Ubuntu with `systemd-networkd`/`systemd-resolved`, where still all relevant network info can be gathered via `ip` and `/etc/resolv.conf`.

Shall we have a common/generic section about gathering the current expectedly DHCP-assigned network info, and then only the second part split for different distros/network stacks, about how to apply the gathered network info statically?